doc/internals/chunked-arrays.rst

```diff
@@ -0,0 +1,91 @@
+.. currentmodule:: xarray
+
+.. _internals.chunkedarrays:
+
+Alternative chunked array types
+===============================
+
+.. warning::
+
+    This is a *highly* experimental feature. Please report any bugs or other difficulties on `xarray's issue tracker <https://github.com/pydata/xarray/issues>`_.
+    In particular see discussion on `xarray issue #6807 <https://github.com/pydata/xarray/issues/6807>`_
+
+Xarray can wrap chunked dask arrays (see :ref:`dask`), but can also wrap any other chunked array type that exposes the correct interface.
+This allows us to support using other frameworks for distributed and out-of-core processing, with user code still written as xarray commands.
+
+The basic idea is that by wrapping an array that has an explicit notion of ``.chunks``, xarray can expose control over
+the choice of chunking scheme to users via methods like :py:meth:`DataArray.chunk` whilst the wrapped array actually
+implements the handling of processing all of the chunks.
+
+Chunked array requirements
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+A chunked array needs to meet all the :ref:`requirements for normal duck arrays <internals.duckarrays>`, but should also implement these methods:
+
+- ``.chunk``
+- ``.rechunk``
+- ``.compute``
+
+Chunked operations as "core operations"
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Xarray dispatches chunk-aware computations across arrays using "core operations" that accept one or more arrays.
+Examples include ``map_blocks``, ``blockwise``, and ``apply_gufunc``.
+These core operations are generalizations of functions first implemented in :py:class:`dask.array`.
+The implementation of these functions is specific to the type of arrays passed to them: :py:class:`dask.array.Array` objects
+must be processed by :py:func:`dask.array.map_blocks`, whereas :py:class:`cubed.Array` objects must be processed by :py:func:`cubed.map_blocks`.
```

Comment:
Need to add `cubed` here: https://github.com/pydata/xarray/blob/77e2db5aa1d8828bc7310062680ad9fe15a65d8a/doc/conf.py#L315